### PR TITLE
PY3 extension upgrade

### PR DESCRIPTION
**Pull request by nickumia-reisys**
PR for https://github.com/GSA/datagov-deploy/issues/3388

### CODE REVIEW

File: ckanext/datagov_inventory/plugin.py [hunk 1]
```diff
@@ -39,8 +44,8 @@ def get_auth_functions(self):
                 'organization_list': datagov_disallow_anonymous_access(),
                 'package_list': datagov_disallow_anonymous_access(),
                 'package_search': datagov_disallow_anonymous_access(),
-                'revision_list': datagov_disallow_anonymous_access(),
-                'revision_show': datagov_disallow_anonymous_access(),
+                'package_show': datagov_disallow_anonymous_access(),
+                'resource_show': datagov_allow_anonymous_access,
```
Comment (jbrown-xentity) [inline ckanext/datagov_inventory/plugin.py:48]: I think we just want to add package_show and resource_show, right?

File: e2e/cypress/integration/dataset.spec.js [hunk 2]
```diff
@@ -40,11 +41,12 @@ describe('Dataset', () => {
...
             formData.set('resource_type', "CSV");
             formData.set('format', "CSV");
-            cy.form_request('POST', 'http://app:5000/api/action/resource_create', formData, function (response) {
+            cy.form_request('POST', '/api/action/resource_create', formData, function (response) {
                 expect(response.status).to.eq(200);
             });
         });
-    })
+        cy.visit('/dataset');
+    });
```
Comment (jbrown-xentity) [inline e2e/cypress/integration/dataset.spec.js:49]: Do we still want this line here?

File: e2e/cypress/integration/public_access.spec.js [hunk 3]
```diff
@@ -14,18 +15,19 @@ describe('Public Access', () => {
...
         cy.delete_organization('test-organization')
     })
 
     it('Cannot access the standard public pages', () => {
+        cy.logout();
         cy.request({
             url: '/dataset',
             failOnStatusCode: false
         }).then((response) => {
             // TODO: local extension should return 403 on anonymous access
```
Comment (jbrown-xentity) [inline e2e/cypress/integration/public_access.spec.js:29]: We can delete this line too!

File: e2e/cypress/integration/login.spec.js [hunk 4]
```diff
@@ -1,15 +1,12 @@
 describe('Login', () => {
     
     it('Invalid user login attempt', () => {
-        cy.visit('/dataset')
-        cy.get('a[href="/user/login"]').click()
+        cy.visit('/user/login')
```
Comment (jbrown-xentity) [inline e2e/cypress/integration/login.spec.js:4]: Don't need this, it's handled in the `cy.login` command

File: config/development.ini [hunk 5]
```diff
@@ -130,8 +135,8 @@ solr_url = http://solr:8983/solr/inventory
 # Note: Add ``datastore`` to enable the CKAN DataStore
 #               Add ``resource_proxy`` to enable resource proxying and get around the
 #               same origin policy
-# TODO: re-integrate datagov_inventory, dcat_usmetadata, usmetadata, datajson, saml2auth
-ckan.plugins = datastore xloader stats text_view recline_view googleanalyticsbasic s3filestore envvars
+# TODO: re-integrate dcat_usmetadata, usmetadata, datajson, saml2auth
+ckan.plugins = datagov_inventory datastore xloader stats text_view recline_view googleanalyticsbasic s3filestore envvars
```
Comment (jbrown-xentity) [inline config/development.ini:139]: This is all good, but it's not added to production.ini

**Review by jbrown-xentity — COMMENTED** [review]
Looking good, but a few more things to do.
We need to get the extension installed in cloud.gov; we can either install it in the cfstart script (my preference) or add it to the requirements install.
We'll also need to enable the extension in the production.ini file.

File: e2e/cypress/integration/public_access.spec.js (hunk 3 above)
Comment (jbrown-xentity) [reply e2e/cypress/integration/public_access.spec.js:29]: By "this", I mean this line: `// TODO: local extension should return 403 on anonymous access`

**Comment by jbrown-xentity** [comment]
Also, would be nice if we could pin pygments to pygments@2.7.4 in the dev requirements, snyk has multiple vulnerabilities with that (it's pulled in via sphinx and ipdb). Or try unpinning sphinx and ipdb... This may break python2/ckan2.8, but `main` will now only support python3/ckan2.9, so I think that's ok if it does.

File: e2e/cypress/integration/dataset.spec.js (hunk 2 above)
Comment (nickumia-reisys) [reply e2e/cypress/integration/dataset.spec.js:49]: It was to make sure the fixture was captured in the right test.  Without this line, the fixture was just run as part of the next test.  I don't know if we can write the test differently to capture the fixture properly.

File: e2e/cypress/integration/public_access.spec.js (hunk 3 above)
Comment (nickumia-reisys) [reply e2e/cypress/integration/public_access.spec.js:29]: 👍

File: config/development.ini (hunk 5 above)
Comment (nickumia-reisys) [reply config/development.ini:139]: 👍

**Review by jbrown-xentity — APPROVED** [review]

**Pull request merged**